src/io/flutter/run/common/CommonTestConfigUtils.java

```diff
@@ -84,21 +84,22 @@ private void clearCachedInfo() {
     }
 
     final FlutterOutline outline = outlineService.getIfUpdated(file);
+    final String path = file.getVirtualFile().getPath();
     // If the outline is outdated, then request a new pass to generate line markers.
     if (outline == null) {
-      clearCachedInfo();
+      clearCachedInfo(path);
       outlineService.addListener(getListenerForFile(file));
       return new HashMap<>();
     }
 
     synchronized (this) {
       // Visit the fields on the outline to get which calls are actual named tests.
-      if (cachedCallToTestType == null) {
-        cachedCallToTestType = new HashMap<>();
-        populateTestTypeMap(outline, cachedCallToTestType, file);
+      if (!cachedCallToTestType.containsKey(path)) {
```

Comment:
what if the outline has changed? we need to clear out the cache in that case.

Reply:
The outline given will be null and the cache will be cleared.

CF: the comments above and the documentation for outlineService.getIfUpdated:

https://github.com/flutter/flutter-intellij/blob/f69eec4dfa31dcd3964fa79d925474376d24e2ad/src/io/flutter/editor/ActiveEditorsOutlineService.java#L187-L194

Reply:
that is not a safe assumption.
The only assumption safe to depend on is the outline will be null if it is out of date.
For example, imagine if another service such as the the widget indent guides or HotUI is also requesting outlines using ActiveEditorsOutlineService, you could easily get a non-null outline even though it is different than the outline your cache is based on.

Reply:
This code checks if the outline is outdated and returns null if so.

If you have suggestions for how to improve the heuristic, they are welcome.

Reply:
the getIfUpdated method is fine as written. What you need to do is not depend on getting a null outline for the case that there was an update. To do that you should store what the outline was in CommonTestConfigUtils so that you can tell if you have gotten a new outline so need to update your cache.

Reply:
Done.